qutebrowser-consult.el

```diff
@@ -0,0 +1,238 @@
+    (consult--multi
+     (mapcar #'qutebrowser-consult--shorten-source-name
+	     (mapcar #'qutebrowser-consult--suppress-action sources))
+     :prompt (if default
+                 (format "Select (default %s): " default)
+               "Select: ")
+     :default default
+     :sort nil
+     :initial initial
+     :require-match nil)))
+
+;;;###autoload
+(defun qutebrowser-consult-launcher (&optional initial target)
+  "Select a URL to open in Qutebrowser using Consult.
+
+Set initial completion input to INITIAL. Open the URL in TARGET or the
+default target if nil.
+
+Modify `qutebrowser-consult-launcher-sources' to change which sources
+are included."
+  (interactive)
+  (let* ((consult-async-split-style nil)
+	 (qutebrowser-default-open-target (or target qutebrowser-default-open-target))
+	 (selected (consult--multi
+		    (mapcar #'qutebrowser-consult--shorten-source-name
+			    (mapcar #'eval qutebrowser-consult-launcher-sources))
+		    :initial initial
+		    :sort nil)))
+    (unless (plist-get (cdr selected) :match)
+      (qutebrowser (car selected)))))
+
+;;;###autoload
+(defun qutebrowser-consult-launcher-tab (&optional initial)
+  "Select a URL to open in a new tab using Consult.
+
+Set initial completion input to INITIAL.
+Modify `qutebrowser-consult-launcher-sources' to change which sources
+are included."
+  (interactive)
+  (qutebrowser-consult-launcher initial 'tab))
+
+;;;###autoload
+(defun qutebrowser-consult-launcher-window (&optional initial)
+  "Select a URL to open in a new window using Consult.
+
+Set initial completion input to INITIAL.
+Modify `qutebrowser-consult-launcher-sources' to change which sources
+are included."
+  (interactive)
+  (qutebrowser-consult-launcher initial 'window))
+
+;;;###autoload
+(defun qutebrowser-consult-launcher-private (&optional initial)
+  "Select a URL to open in a private window using Consult.
+
+Set initial completion input to INITIAL.
+Modify `qutebrowser-consult-launcher-sources' to change which sources
+are included."
+  (interactive)
+  (qutebrowser-consult-launcher initial 'private-window))
+
+(define-minor-mode qutebrowser-consult-mode
+  "Use Consult for Qutebrowser completion."
+  :lighter nil
+  :global t
+  (if qutebrowser-consult-mode
+      (progn
+	(advice-add 'qutebrowser-launcher :override #'qutebrowser-consult-launcher)
+	(advice-add 'qutebrowser-launcher-tab :override #'qutebrowser-consult-launcher-tab)
+	(advice-add 'qutebrowser-launcher-window :override #'qutebrowser-consult-launcher-window)
+	(advice-add 'qutebrowser-launcher-private :override #'qutebrowser-consult-launcher-private))
+    (advice-remove 'qutebrowser-launcher #'qutebrowser-consult-launcher)
+    (advice-remove 'qutebrowser-launcher-tab #'qutebrowser-consult-launcher-tab)
+    (advice-remove 'qutebrowser-launcher-window #'qutebrowser-consult-launcher-window)
+    (advice-remove 'qutebrowser-launcher-private #'qutebrowser-consult-launcher-private)))
```

Comment:
This is actually also a quite nice way to handle it. But I think I still prefer the approach I described above.